File: observer/java/IObserverUser.java
package Observer;

// Değişimden etkilenecek, izleyecek olan sınıfların uyguladığı arayüzdür.
// UML diyagramındaki Observer yapısına denk gelmektedir.
interface IObserverUser {
    void sendNotification(ProductUpdateMessage message);
}

File: observer/kotlin/IObserverUser.kt
package Observer

// Değişimden etkilenecek, izleyecek olan sınıfların uyguladığı arayüzdür.
// UML diyagramındaki Observer yapısına denk gelmektedir.
internal interface IObserverUser {
    fun sendNotification(message: ProductUpdateMessage)
}